modules/packages/LinearAlgebra.chpl

```diff
@@ -2313,37 +2313,33 @@ module Sparse {
 
   /* Return a CSR matrix constructed from internal representation:
 
-    - ``shape``: bounding box dimensions
+    - ``shape``: index ranges for rows and columns
     - ``data``: non-zero element values
     - ``indices``: non-zero row pointers
     - ``indptr``: index pointers
 
   */
-  proc CSRMatrix(shape: 2*int, data: [?nnzDom] ?eltType, indices: [nnzDom], indptr: [?indDom])
-    where indDom.rank == 1 && nnzDom.rank == 1 {
+  // ?fromMMS: do we need to deprecate the shape: 2*int version of this
+  proc CSRMatrix(shape: 2*range, data: [?nnzDom] ?eltType, indices: [nnzDom], indptr: [?indDom])
```

Comment:
I suggest we add a new variant:

```chpl
proc CSRMatrix(parentDom: domain(2),  data: [?nnzDom] ?eltType, indices: [nnzDom], indptr: [?indDom])
```

... then add a deprecation warning from the existing implementation.

Deprecation involves:

- adding a compiler warning to the deprecated routine
- adding a test for the deprecation warning
- stating that the routine is deprecated in the docs, typically with a `.. warning::` block.

Details can be found in the [deprecation doc](https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/bestPractices/Deprecation.rst)

Reply:
Did as recommended in the sparse-0based branch.